### ISSUE TITLE
Internal networking does not work with encryption enabled until a rollout restart is performed

### ISSUE THREAD

**Issue by msarti** [cilium/cilium#19958]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

After installation of Cilium on a newly bootstrapped cluster with wireguard encryption enabled, the pods do not reach any addresses within the cluster. Issue does not appear if I do not enable encryption.
It is possible to get around the issue with a rollout restart of the cilium daemonset.

values.yaml:
```
kubeProxyReplacement: strict
l7Proxy: false
encryption:
  enabled: true
  type: wireguard
ipam:
  mode: "cluster-pool"
  operator:
    clusterPoolIPv4PodCIDRList: "10.244.0.0/16"
    clusterPoolIPv4MaskSize: "24"
hostServices:
  enabled: true
k8sServiceHost: 192.168.0.100
k8sServicePort: 8443
```
To reproduce the issue:

```
kubectl apply -f https://k8s.io/examples/admin/dns/dnsutils.yaml

kubectl exec -i -t dnsutils -- nslookup kubernetes.default
;; connection timed out; no servers could be reached

command terminated with exit code 1

```
To work around the issue:

```
kubectl -n kube-system rollout restart ds/cilium

kubectl exec -i -t dnsutils -- nslookup kubernetes.default
Server:		10.97.0.10
Address:	10.97.0.10#53

Name:	kubernetes.default.svc.cluster.local
Address: 10.97.0.1
```

### Cilium Version

```
Client: 1.11.5 b0d3140 2022-05-10T02:47:42+02:00 go version go1.17.9 linux/amd64
Daemon: 1.11.5 b0d3140 2022-05-10T02:47:42+02:00 go version go1.17.9 linux/amd64

```

### Kernel Version

`Linux default-cp-01 5.15.0-30-generic #31-Ubuntu SMP Thu May 5 10:00:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`

### Kubernetes Version

```
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-03T13:46:05Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-03T13:38:19Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}

```

### Relevant log output

```shell
No relevant logs detected.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by kvaster** [comment]
It seems that my problem is related to NodeGC - disabling node gc fixes the problem.

Also I had non-default params for k8s controller:

```
node-monitor-grace-period: 16s
node-monitor-period: 2s
pod-eviction-timeout: 30s
```

I decided to make some experiments and removed this params (without disabling NodeGC in cilium). Problem was gone in 90% cases. After that I've applied following params:

```
node-startup-grace-period: 5m0s
node-monitor-grace-period: 2m0s
```

And there is no more problem with such config...
